google/cloud/aiplatform/training_jobs.py

```diff
@@ -616,181 +908,233 @@ def run(
-                training_fraction=training_fraction_split,
-                validation_fraction=validation_fraction_split,
-                test_fraction=test_fraction_split,
-            )
-
-            # create input data config
-            input_data_config = gca_training_pipeline.InputDataConfig(
-                fraction_split=fraction_split,
-                dataset_id=dataset.name,
-                gcs_destination=gca_io.GcsDestination(
-                    output_uri_prefix=base_output_dir
-                ),
-            )
-
-        # create training pipeline
-        training_pipeline = gca_training_pipeline.TrainingPipeline(
-            display_name=self._display_name,
-            training_task_definition=schema.training_job.definition.custom_task,
-            training_task_inputs=json_format.ParseDict(
-                {
-                    "workerPoolSpecs": [worker_pool_spec],
-                    "baseOutputDirectory": {"output_uri_prefix": base_output_dir},
-                },
-                struct_pb2.Value(),
-            ),
-            model_to_upload=managed_model,
-            input_data_config=input_data_config,
+        return super()._run_job(
+            training_task_definition=training_task_definition,
+            training_task_inputs=training_task_inputs,
+            dataset=dataset,
+            training_fraction_split=training_fraction_split,
+            validation_fraction_split=validation_fraction_split,
+            test_fraction_split=test_fraction_split,
+            model=managed_model,
+            base_output_dir=base_output_dir,
         )
 
-        training_pipeline = self.api_client.create_training_pipeline(
-            parent=initializer.global_config.common_location_path(
-                self.project, self.location
-            ),
-            training_pipeline=training_pipeline,
+    @property
+    def _model_upload_fail_string(self) -> str:
+        """Helper property for model upload failure."""
+        return (
+            f"Training Pipeline {self.resource_name} is not configured to upload a "
+            "Model. Create the Training Pipeline with "
+            "model_serving_container_image_uri and model_display_name passed in. "
+            "Ensure that your training script saves to model to "
+            "os.environ['AIP_MODEL_DIR']."
         )
 
-        self._gca_resource = training_pipeline
 
-        _LOGGER.info("View Training:\n%s" % self._dashboard_uri())
-        _LOGGER.info("Training Output directory:\n%s " % base_output_dir)
-
-        model = self._get_model()
-
-        if model is None:
-            _LOGGER.warn(
-                "Training did not produce a Managed Model returning None. "
-                + self._model_upload_fail_string
-            )
-        return model
+class AutoMLTablesTrainingJob(_TrainingJob):
+    def __init__(
+        self,
+        display_name: str,
+        optimization_objective: str,
```

Comment:
According to the protos and yaml, the `optimization_objective` is optional since there is a default if it's not supplied.

Also, the design doc has this signature, which doesn't have the `optimization_prediction_type` parameter either. We can support this by inferring that info based on the `optimization_objective`, but that can be done a multitude of ways. 

My personal preferred way is to create an OptimizationObjective abstract class and create subclasses of each for each `optimization_objective` type. That would encapsulate the `optimization_objective`, `optimization_prediction_type `, `optimization_objective_recall_value` and `optimization_objective_precision_value` parameters together into one. 

However, from our other convos it seems like you might prefer just passing in a string for `optimization_objective` and having the other parameters be optional.

Let me know what you prefer.

Reply:
@sasha-gitg

Reply:
With respect to `optimization_prediction_type` we received feedback to include that as the top level argument because it requires less knowledge overhead than `optimization_objective`. Additionally, we should include all the values flattened in the API surface and validate or ignore arguments based on valid combinations. There's precedence for this pattern:

https://github.com/scikit-learn/scikit-learn/blob/0fb307bf3/sklearn/linear_model/_logistic.py#L1011

So the current state of the input arguments LGTM just elevate `optimization_prediction_type` over `optimization_objective` and make `optimization_objective` optional but add comments to explain the defaults when selecting `optimization_prediction_type`.

Reply:
Thanks for the context, sgtm!